src/components/FormModal/FormModal.js

```diff
@@ -92,27 +98,100 @@ export default function FormModal({ onCancelAdding, id, name }) {
   }
 
   function handleMorningClicked(name) {
-    const newMorningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.morning = !day.morning;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedMornings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].morning === true
+    );
+
+    const routineIDs = checkedMornings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
     });
-    setWeekRoutine({ id: id, days: newMorningChecked, date: weekRoutine.date });
+
+    if (intersection.length > 0) {
```

Comment:
I see, that you are using the intersection list for some conditions here. In my change suggestion above I deleted it. May change the changing in a way, that this code is working too. Sorry about that.